### ISSUE TITLE
Failed on parsing jquery (not minimized) - "parsererror"

### ISSUE THREAD

**Issue by avakarev** [veged/ometa-js#6]
It seems parser doesn't like "parsererror" token in the source.

Here is what I've got:

```
$:> node test/benchmark/bsjs.js
Source length: 252881

/Users/avakarev/Development/oss/ometa-js/lib/ometajs/core/grammar.js:46
    if (!errback) errback = function (err) { throw err; };
                                                   ^
Error: str rule failed at: 596:88
        if ( !xml || !xml.documentElement || xml.getElementsByTagName( "parsererror" ).length ) {
                                                                                        ^
    at Function.matchAll (/Users/avakarev/Development/oss/ometa-js/lib/ometajs/core/grammar.js:53:39)
    at Object.<anonymous> (/Users/avakarev/Development/oss/ometa-js/test/benchmark/bsjs.js:12:8)
    at Module._compile (module.js:446:26)
    at Object..js (module.js:464:10)
    at Module.load (module.js:353:31)
    at Function._load (module.js:311:12)
    at Array.0 (module.js:484:10)
    at EventEmitter._tickCallback (node.js:190:38)
```

Just interesting: why default branch is nodejs instead of master? :-)

FYI: anything I've tested is in branch in my fork: https://github.com/avakarev/ometa-js/tree/testing-to-parse-popular-libs

**Comment by avakarev** [comment]
So, I've been playing a little with it. I've comment out all the lines in jquery which confused ometa-js. 
There are around 34 patches. 

Of course, jquery will not work correctly now but it became parsable! (changes in branch, I've pointed above)

Is it correct to compare the performance of ometa-js against jshint's parser?

```
$:> time node test/benchmark/bsjs.js                  
Source length: 253175
parse: 5292ms
node test/benchmark/bsjs.js  5.18s user 0.25s system 99% cpu 5.482 total
[~/Development/oss/ometa-js] @MacBook-admin
$:> time jshint test/benchmark/samples/jquery-1.7.2.js
jshint test/benchmark/samples/jquery-1.7.2.js  0.42s user 0.03s system 96% cpu 0.468 total
```

**Comment by indutny** [comment]
Fixed parser errors in 3.1.2 version of ometajs. Though, parse time is still very high in comparison to other parsers. Unfortunatelly, it seems that this limitation is a consequence of having extensible grammar. I'll look more into this issue, but can't promise any performance improvements.

**Comment by indutny** [comment]
I've tuned performance a little, parsing jquery now lasts 3086ms on my MBP 15''
